File: Java/src/main/java/QuestionBank/Solution101_200/Solution104.java
package QuestionBank.Solution101_200;

import tool.TreeNode;

/**
 * @author windows
 */
public class Solution104 {
    public static void main(String[] args) {
        Solution104 solution104 = new Solution104();
    }

    public int maxDepth(TreeNode root) {
        if (root == null) {
            return 0;
        }
        int left = 0;
        int right = 0;
        left = maxDepth(root.left) + 1;
        right = maxDepth(root.right) + 1;
        return Math.max(left, right);
    }
}

File: C++/src/Solution101_200/Solution104.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution104.h"
#include <algorithm>
using namespace std;

int Solution104::maxDepth(TreeNode *root) {
    return dfs(root, 0);
}

int Solution104::dfs(TreeNode *root, int depth) {
    if (root == nullptr){
        return depth;
    }
    int left = dfs(root->left, depth + 1);
    int right = dfs(root->right, depth + 1);
    return max(left, right);
}

